### ISSUE TITLE
Postinstall script fails in 5.0.0 with ECONNRESET

### ISSUE THREAD

**Issue by Soulusions** [prisma/prisma#20193]
### Bug description

I just upgraded a working project to prisma 5 and now simply trying to run `pnpm install` causes an error as follows:
```
node_modules/.pnpm/@prisma+engines@5.0.0/node_modules/@prisma/engines: Running postinstall script, failed in 50.7s
.../node_modules/@prisma/engines postinstall$ node scripts/postinstall.js
│ node:internal/process/promises:288
│             triggerUncaughtException(err, true /* fromPromise */);
│ Error: read ECONNRESET
│     at TLSWrap.onStreamRead (node:internal/stream_base_commons:217:20) {
│   errno: -4077,
│   code: 'ECONNRESET',
│   syscall: 'read',
│   attemptNumber: 1,
│   retriesLeft: 2
│ }
│ Node.js v18.16.0
```

### How to reproduce

1. Set prisma CLI version as `^5.0.0`
2. Run `pnpm i`

### Expected behavior

Postinstall script should succeed

### Prisma information

```prisma
model User {
  id    Int     @id @default(autoincrement())
  email String  @unique
  name  String?
  posts Post[]
}

model Post {
  id        Int     @id @default(autoincrement())
  title     String
  content   String?
  published Boolean @default(false)
  author    User    @relation(fields: [authorId], references: [id])
  authorId  Int
}
```

### Environment & setup

- OS: Windows 11
- Database: pgsql
- Node.js version: 18.16.0

### Prisma Version

```
5.0.0
```

**Comment by Jolg42** [comment]
@Soulusions Could you try 
- Running `DEBUG=* pnpm install`?
- Running `DEBUG=* pnpm prisma -v`?
- Downloading manually https://binaries.prisma.sh/all_commits/6b0aef69b7cdfc787f822ecd7cdc76d5f1991584/windows/schema-engine.exe.gz to see if that works?

**Comment by Jolg42** [comment]
Note: this looks like a temporary network issue to us, so retrying might just work.

**Comment by Soulusions** [comment]
Right, after further investigation this seemed to only be a problem when running the install command in the vsc cli, from a regular old terminal everything went fine, which is quite odd but whatever, thanks for the pointer though!

**Comment by luizfelipelaviola** [comment]
I got the same error using the terminal inside VSCode. Using the terminal out of VSCode works. 🤷🏼‍♂️

**Comment by janpio** [comment]
Also on Windows as @Soulusions?

**Comment by Dominguezd01** [comment]
Its happening to me trying to deploy on Render.com

**Comment by janpio** [comment]
Can you reproduce this with a new, minimal project created with `npx prisma init` @Dominguezd01?
If so, we could probably try to reproduce this on Render ourselves.

**Comment by Dominguezd01** [comment]
I create the project with npx prisma init and uploaded to Render with different error, i link you the github repo to check it out (im using neon.tech as DB provider)

https://github.com/Dominguezd01/PrismaTest

I hope this helps, peace

**Comment by Jolg42** [comment]
@Dominguezd01 I created a separate issue from your comment, and I'm going to hide the unrelated comments to keep this issue clean ✨ .
See https://github.com/prisma/prisma/issues/20260

**Comment by aqrln** [comment]
```
│   attemptNumber: 1,
│   retriesLeft: 2
```
Looks concerning, from the logs it seems like the automatic retrying might not work and threw an error prematurely

**Comment by Jolg42** [comment]
@Soulusions I improved the logic around that code in https://github.com/prisma/prisma/pull/20419

Could you try this special version? See below:

On Windows you can run
```
set DEBUG=* && npx prisma@5.1.0-integration-joel-improve-download.3
```
And then if you could share the output here that would help. Thanks!

**Comment by Jolg42** [comment]
@Soulusions Could you try again using `5.1.1`?
```
set DEBUG=* && npx prisma@5.1.1
```

**Comment by Jolg42** [comment]
Closing this for now as stale, if the problem persists, please let us know and check the previous messages.

**Comment by ofirgaash1** [comment]
solved for me by connecting to another wifi (from phone hotspot)
